src/__tests__/views/resetView/__test__/resetPasswordView.test.js

```diff
@@ -0,0 +1,117 @@
+import React from "react";
+import { shallow, mount } from "enzyme";
+import { Provider } from "react-redux";
+import configureStore from "redux-mock-store";
+import ResetPasswordView, { 
+  ResetPasswordView as ResetPasswordViewTest,
+} from "../../../../views/resetView/resetPasswordView";
+
+const initialState = {
+  resetPasswordReducer: {
+    message: "hello",
+  },
+};
+const mockStore = configureStore();
+const store = mockStore(initialState);
+
+const props = {
+  resetPasswordAction: jest.fn(),
+  match: {
+    params: {
+      token: "ghjkbvfghj",
+    },
+  },
+  changePasswordAction: jest.fn(),
+  history: { push: jest.fn() },
+};
+
+const initialProps = {
+  resetPasswordAction: jest.fn(),
+  match: {
+    params: {
+
+    },
+  },
+};
+
+describe("ResetPasswordView", () => {
+  const wrapper1 = mount(
+    <Provider store={store}>
+      <ResetPasswordView {...props} />
+    </Provider>,
+  );
+  it("should render without crashing", () => {
+    expect(wrapper1).toMatchSnapshot();
+  });
+
+  let wrapper;
+  let instance;
+
+  beforeEach(() => {
+    wrapper = shallow(
+      <ResetPasswordViewTest {...props} />,
+    );
+    instance = wrapper.instance();
+  });
+  it("should redirect on new message", () => {
+    const email = "testemail@gmail.com";
+    const wrapper = shallow(<ResetPasswordView {...props} />);
+    wrapper.instance().props.resetPasswordAction(email);
+    expect(wrapper.instance().props.resetPasswordAction).toBeCalled();
+  });
+
+  it("should call changePasswordAction when correct password data has been passed", () => {
+    wrapper.setState({ password: "aciopeace", confirmPassword: "aciopeace" });
+    instance.handlePasswordSubmit({
+      preventDefault: jest.fn(),
+    });
+    expect(props.changePasswordAction).toHaveBeenCalled();
+  });
+
+  it("should fail when unmatching passwords are entered", () => {
+    wrapper.setState({ password: "acio", confirmPassword: "acionpeace" });
+    instance.handlePasswordSubmit({
+      preventDefault: jest.fn(),
+    });
+  });
+
+  it("should handle onChange with correct email input", () => {
+    instance.onChange({ target: { name: "email", value: "apple@acio.com" } });
+  });
+
+  it("should handle onChange with wrong email input", () => {
+    instance.onChange({ target: { name: "email", value: "appleacio.com" } });
+  });
+    
+  it("should call component will receive props", () => {
+    wrapper.setProps({ message: "fake message" });
+    instance.componentWillReceiveProps({
+      preventDefault: jest.fn(),
+    });
+    // expect(props.componentWillReceiveProps).toHaveBeenCalled();
```

Comment:
@Peace-Apple please remove this commented out code

Reply:
Thanks, I have worked on that